src/resolve-config.js

```diff
@@ -9,7 +9,13 @@ const getExplorerMemoized = mem(opts =>
   cosmiconfig("prettier", {
     sync: opts.sync,
     cache: opts.cache,
-    rcExtensions: true
+    rcExtensions: true,
+    transform: result => {
+      if (result && result.config) {
+        delete result.config.$schema;
+      }
+      return opts.sync ? result : Promise.resolve(result);
```

Comment:
It seems we do not need to handle Promise ourselves:

https://github.com/davidtheclark/cosmiconfig/blob/930926e7ca4e4f4c6cd57c371748a6c9d45083d8/src/createExplorer.js#L90-L93

Reply:
Ah, cool. I might send a PR to cosmiconfig to clarify that in the docs